frontend/src/pages/projects/screens/spawner/storage/ClusterStorageEditModal.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+
+import {
+  Alert,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+  TextArea,
+  TextInput,
+} from '@patternfly/react-core';
+
+import DashboardModalFooter from '~/concepts/dashboard/DashboardModalFooter';
+import PVSizeField from '~/pages/projects/components/PVSizeField';
+import { StorageData } from '~/pages/projects/types';
+import FieldGroupHelpLabelIcon from '~/components/FieldGroupHelpLabelIcon';
+import StorageClassSelect from './StorageClassSelect';
+
+interface ClusterStorageEditModalProps {
+  storageData: StorageData;
+  onUpdate: (storageData: StorageData) => void;
+  onClose: () => void;
+}
+
+export const ClusterStorageEditModal: React.FC<ClusterStorageEditModalProps> = ({
+  storageData,
+  onUpdate,
+  onClose,
+}) => {
+  const [name, setName] = React.useState(storageData.name);
+  const [description, setDescription] = React.useState(storageData.description);
+  const [size, setSize] = React.useState(storageData.size);
+  const [storageClassName, setStorageClassName] = React.useState(storageData.storageClassName);
+  const [mountPath, setMountPath] = React.useState(storageData.mountPath);
+
+  const onSubmit = () => {
+    onUpdate({
+      ...storageData,
+      name,
+      description,
+      size,
+      storageClassName,
+      mountPath,
+    });
+    onClose();
+  };
+
+  return (
+    <Modal
+      isOpen
+      variant={ModalVariant.small}
+      title="Edit storage"
+      onClose={onClose}
+      footer={
+        <DashboardModalFooter
+          onCancel={onClose}
+          onSubmit={onSubmit}
+          submitLabel="Save"
+          isSubmitDisabled={!name}
```

Comment:
Is the submit button disabled status check also implemented in the other ticket?

Reply:
As shared [here](https://github.com/opendatahub-io/odh-dashboard/pull/3320#discussion_r1801825868), this is just a start to the modal, so yes